src/Library/demos/Welcome/main.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+import gi
+
+gi.require_version("Gtk", "4.0")
+gi.require_version("Adw", "1")
+from gi.repository import Gtk, Adw
+
+# TODO: The previewer process has to inject workbench.py as a module into sys.path later.
+import workbench
+
+
+def main():
+    subtitle_box: Gtk.Box = workbench.builder.get_object("subtitle")
+    button: Gtk.Button = Gtk.Button.new_with_label("Press me")
+    button.set_margin_top(6)
+    button.add_css_class("suggested-action")
+
+    button.connect("clicked", greet)
+    subtitle_box.append(button)
+
+    print("Welcome to Workbench!")
+
+
+def greet():
+    dialog: Adw.MessageDialog = Adw.MessageDialog.new(body="Hello World!")
+    dialog.set_transient_for(workbench.window)
+
+    dialog.add_response("ok", "Ok")
+    dialog.connect("response", handle_response)
+    dialog.present()
+
+
+def handle_response(dialog: Adw.MessageDialog, response: str):
+    print(response)
+    dialog.close()
+
+
+if __name__ == "__main__":
```

Comment:
Not really, it's good convention to have this in Python scripts that are meant to be run from CLI since this will only evaluate to `True` if run as the main script. Everything on top-level will pollute the module namespace.

But just omitting it and the main function would simplify the example a bit. And depending on how we end up invoking the scripts this approach may not work.